luigi/contrib/jupyter_notebook.py

```diff
@@ -0,0 +1,310 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module is intended for when you need to execute a Jupyter notebook as a
+task within a Luigi pipeline. This can be accomplished via the
+:class:`JupyterNotebookTask` class.
+
+:class:`JupyterNotebookTask` allows you to pass parameters to the Jupyter
+notebook through the ``pars`` dictionary.
+
+When the task is executed, the ``pars`` dictionary is written to a
+temporary JSON file in the same directory that contains the notebook.
+
+Inside the notebook, you can retrieve the values of the parameters in
+``pars`` by reading the temporary JSON file. The path to the temporary file is
+accessible via the environment variable ``PARS``.
+For example, in a Python notebook, you can read the ``pars`` dictionary as
+follows:
+
+.. code-block:: python
```

Comment:
These example code blocks are very nice to have. I would add another block with an example of how to run the task. This could be along the lines you sent me previously. In particular, 

1. Look up the available kernels with `jupyter kernelspec list` (which usually correspond to the Python versions).
2. After starting `luigid` run:

```
luigi --module jupyter_notebook JupyterNotebookTask --nb-path=./my_notebook.ipynb --kernel-name=python3
```

Reply:
I will shortly add some more documentation containing an example of how to run the task. You're right, that's a good idea :)